Sources/OLEKit/OLEFile.swift

```diff
@@ -40,26 +40,46 @@ public final class OLEFile {
     guard let fileHandle = FileHandle(forReadingAtPath: path)
     else { throw OLEError.fileNotAvailableForReading(path: path) }
 
-    self.fileHandle = fileHandle
+    let allData = fileHandle.readDataToEndOfFile()
+    fileHandle.seek(toFileOffset: UInt64(0))
 
+    try self.init(data: allData, fileSize: fileSize, path: path)
+  }
+
+  #if os(iOS) || os(watchOS) || os(tvOS) || os(macOS)
+
+  public convenience init(_ fileWrapper: FileWrapper) throws {
+    let fileName = fileWrapper.filename ?? ""
+
+    guard let data = fileWrapper.regularFileContents
+    else { throw OLEError.fileDoesNotExist(fileName) }
+
+    guard let fileSize = fileWrapper.fileAttributes[FileAttributeKey.size.rawValue] as? Int
+    else { throw OLEError.fileDoesNotExist(fileName) }
```

Comment:
If the same error is thrown from both conditions, it makes sense to group them in a single `guard`.

```suggestion
    guard
      let data = fileWrapper.regularFileContents,
      let fileSize = fileWrapper.fileAttributes[FileAttributeKey.size.rawValue] as? Int
    else { throw OLEError.fileDoesNotExist(fileName) }
```